src/bridges/element/ElementBridge.sol

```diff
@@ -469,47 +475,57 @@ contract ElementBridge is IDefiBridge {
 
         // retrieve the appropriate pool for this interaction and verify that it exists
         Pool storage pool = pools[hashAssetAndExpiry(convertArgs.inputAssetAddress, convertArgs.auxData)];
-        if (pool.trancheAddress == address(0)) {
+        address trancheAddress = pool.trancheAddress;
+        if (trancheAddress == address(0)) {
             revert POOL_NOT_FOUND();
         }
-        ITranche tranche = ITranche(pool.trancheAddress);
+        ITranche tranche = ITranche(trancheAddress);
         uint64 trancheExpiry = uint64(tranche.unlockTimestamp());
         if (block.timestamp >= trancheExpiry) {
             revert TRANCHE_ALREADY_EXPIRED();
         }
-        
+
         // approve the transfer of tokens to the balancer address
-        ERC20(convertArgs.inputAssetAddress).approve(balancerAddress, convertArgs.totalInputValue);
         // execute the swap on balancer
-        uint256 principalTokensAmount = exchangeAssetForTrancheTokens(convertArgs.inputAssetAddress, pool, convertArgs.totalInputValue);
+        uint256 principalTokensAmount = exchangeAssetForTrancheTokens(
+            convertArgs.inputAssetAddress,
+            pool,
+            convertArgs.totalInputValue
+        );
         // store the tranche that underpins our interaction, the expiry and the number of received tokens against the nonce
         Interaction storage newInteraction = interactions[convertArgs.interactionNonce];
-        newInteraction.expiry = trancheExpiry;
-        newInteraction.failed = false;
-        newInteraction.finalised = false;
         newInteraction.quantityPT = principalTokensAmount;
-        newInteraction.trancheAddress = pool.trancheAddress;
+        newInteraction.trancheAddress = trancheAddress;
+        newInteraction.expiry = trancheExpiry;
+        //newInteraction.failed = false;
```

Comment:
We can remove these commented out lines

Reply:
Updated in https://github.com/AztecProtocol/aztec-connect-bridges/pull/67/commits/172d2833533797e11d426f7f4e27eee119bb2046